ibis/backends/dask/client.py

```diff
@@ -244,19 +244,18 @@ def create_table(
         schema: sch.Schema = None,
     ):
         """Create a table."""
-        if obj is None and schema is None:
-            raise com.IbisError('Must pass expr or schema')
-
         if obj is not None:
             df = obj
-        else:
+        elif schema is not None:
```

Comment:
why this change? this is a logic change

Reply:
Not sure if there was any mypy problem in the original code, but the logic is the same (not immediately obvious, took me a bit to see it), and I think the new implementation is clearer.

Reply:
Without this change, mypy raise this error:
```
ibis/backends/dask/client.py:254: error: Item "None" of "Optional[Schema]" has no attribute "apply_to"
```
The logic is the same, but in the original code mypy cannot figure out that schema will never be `None` in that line.

Reply:
Shouldn't this look the same in pandas then? https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/client.py#L409

I'm fine with the change, but I'd expect that the `PandasClient` and `DaskClient` look extremely similar code-wise.

Reply:
I created issues to address one backend at a time. But we'll surely have to make the same change in the pandas backend when we fix the mypy errors on it, since the mypy problem will be the same.

Reply:
@arielrossanigo have taken this issue in the pandas backend, maybe we can coordinate to make the same change in both backends

Reply:
Hi people, I've checked and currently in the `PandasClient` there is no mypy problem because of the lack of annotations. 

I think @sofide can reflect the change in the `PandasClient` and then I rebase my branch when this PR is merged, What do you think?

Reply:
Don't worry about this, this PR should be ready. Will wait a bit more to see if @jreback can have a last look, but all is fine.

And regarding pandas, not so concerned that this function has the same code. I don't think we should have copied pasted anyway, so if we want to do things properly what we should do is use the pandas function for dask, not make sure we make the same changes in both functions. But let's focus on the mypy problems.